torch/csrc/jit/symbolic_variable.h

```diff
@@ -77,11 +80,27 @@ struct SymbolicVariable {
     return create(kneg, {*this})[0].typeLike(*this);
   }
   SymbolicVariable mm(const SymbolicVariable rhs) const {
-    // TODO: set types
-    return create(s("mm"), {*this, rhs})[0];
+    auto r = create(s("mm"), {*this, rhs})[0];
+    if (v->hasType() && rhs.v->hasType()) {
```

Comment:
These type propagations are going to get redundant with the shape propagation pass. Maybe we can move this code into shape propagation, and then expose another function that propagates types on a single node that can be called from here. That way we can reuse this code for both SymbolicVariable, and the shape propagation.

Reply:
Right, I forgot we have that pass. I'll fix it.